File: 2126. Destroying Asteroids.py
class Solution:
    def asteroidsDestroyed(self,
                           initial_mass: int,
                           asteroid_list: List[int]) -> bool:

        # Process asteroids from smallest to largest
        asteroid_list.sort()

        # Current mass of the spaceship
        spaceship_mass = initial_mass

        for asteroid_size in asteroid_list:

            # Cannot destroy a larger asteroid
            if spaceship_mass < asteroid_size:
                return False

            # Gain the asteroid's mass after destroying it
            spaceship_mass += asteroid_size

        return True

File: 2126. Destroying Asteroids.cpp
class Solution {
public:

    bool asteroidsDestroyed(int initialMass,
                            vector<int>& asteroidList) {

        // Process asteroids from smallest to largest
        sort(asteroidList.begin(), asteroidList.end());

        // Use long long to avoid overflow
        long long spaceshipMass = initialMass;

        for (int asteroidSize : asteroidList) {

            // Cannot destroy an asteroid larger than current mass
            if (spaceshipMass < asteroidSize) {
                return false;
            }

            // Destroy asteroid and gain its mass
            spaceshipMass += asteroidSize;
        }

        return true;
    }
};
